src/snowflake/snowpark/async_job.py

```diff
@@ -328,47 +334,51 @@ def result(
             # scan to have Snowflake read the result and return Arrow format.
             # TODO: Once we support fetch_pandas_all for multi-statement query in connector, we could remove this
             #   workaround.
-            if result_type in (_AsyncResultType.PANDAS, _AsyncResultType.PANDAS_BATCH):
+            if async_result_type in (
+                _AsyncResultType.PANDAS,
+                _AsyncResultType.PANDAS_BATCH,
+            ):
                 self._cursor.execute(
                     f"select * from table(result_scan('{self._cursor.sfqid}'))"
                 )
 
-        if result_type == _AsyncResultType.NO_RESULT:
+        if async_result_type == _AsyncResultType.NO_RESULT:
             result = None
-        elif result_type == _AsyncResultType.PANDAS:
+        elif async_result_type == _AsyncResultType.PANDAS:
             result = self._session._conn._to_data_or_iter(
                 self._cursor, to_pandas=True, to_iter=False
             )["data"]
             check_is_pandas_dataframe_in_to_pandas(result)
-        elif result_type == _AsyncResultType.PANDAS_BATCH:
+        elif async_result_type == _AsyncResultType.PANDAS_BATCH:
             result = self._session._conn._to_data_or_iter(
                 self._cursor, to_pandas=True, to_iter=True
             )["data"]
         else:
             result_data = self._cursor.fetchall()
             self._result_meta = self._cursor.description
-            if result_type == _AsyncResultType.ROW:
+            if async_result_type == _AsyncResultType.ROW:
                 result = result_set_to_rows(
                     result_data,
                     self._result_meta,
                     case_sensitive=self._case_sensitive,
                 )
-            elif result_type == _AsyncResultType.ITERATOR:
+            elif async_result_type == _AsyncResultType.ITERATOR:
+                assert isinstance(result_data, SnowflakeCursor)
```

Comment:
Tests are failing on this line. I'm just going to remove it and not worry about the type error, I added this before I decided to only aim for 0 errors under `_internal/analyzer/`. But this means that `def result_set_to_iter(result_set: SnowflakeCursor,` isn't correct.